sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -787,9 +790,9 @@ class RocksDB(
 
   /** Release all resources */
   def close(): Unit = {
+    // Acquire DB instance lock and release at the end to allow for synchronized access
+    acquire(CloseStore)
```

Comment:
Previously we swallowed any exception that happens here (see the catch). By moving this `acquire()` call out, if we fail to acquire lock, which then throws exception, we will not swallow it. Is that the intention here?

Reply:
yes - if we cannot acquire the lock - we should fail the task